### ISSUE TITLE
iconv issue converting 1047 to 819

### ISSUE THREAD

**Issue by gngrossi** [ZOSOpenTools/libiconvport#34]
Commands used to convert the ebcdic file to ascii using iconv and and /bin/iconv
touch <new file>
chtag -tc ISO8859-1 <new file>
iconv -f IBM-1047 -t ISO8859-1 $1.ebcdic > <new file>

t IBM-1047    T=on  -rw-r-----   1 @02858   @ISSTC00     221 Feb 23 16:46 TBASVC_02132024.csv_GG
t ISO8859-1   T=on  -rw-r-----   1 @02858   @ISSTC00     221 Feb 23 16:49 TBASVC_02132024.csv_GG.zopen_iconv
t ISO8859-1   T=on  -rw-r-----   1 @02858   @ISSTC00     221 Feb 23 16:52 TBASVC_02132024.csv_GG.iconv

1047 file
bash-5.2$ cat TBASVC_02132024.csv_GG
"DATE","TRANNAME","CLIENTID","TBAIND","TBAS01","cputm","ttrans","cpuptrn","db2reqct","mqreqct",DB2/tran,SUs/tran,TotalCost
"12FEB24","ABCD",00000,999,"SERVICEABC",12445,428684,0.0290,375813738,249414,876.67,209.61,373.35

819 file using iconv
bash-5.2$ cat TBASVC_02132024.csv_GG.zopen_iconv
▒▒▒ႌ▒▒▒++▒(ႌ▒▒<▒▒+▒▒▒▒▒▒▒+▒▒▒▒됑▒▒▒▒▒▒▒_▒▒▒▒▒▒/>˂▒▒▒▒▒▒▒▒>▒▒▒▒▒▒▒▒▒Ȃ▒▒_▒▒▒▒▒Ȃ▒▒▒▒▒/>▒▒▒▒▒▒/>▒▒?▒/%▒?▒Ȏ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒䂌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒16:56:52 RC=(0) [SYSA] bash-5.2$

819 file using /bin/iconv
bash-5.2$ cat TBASVC_02132024.csv_GG.iconv
"DATE","TRANNAME","CLIENTID","TBAIND","TBAS01","cputm","ttrans","cpuptrn","db2reqct","mqreqct",DB2/tran,SUs/tran,TotalCost
"12FEB24","ABCD",00000,999,"SERVICEABC",12445,428684,0.0290,375813738,249414,876.67,209.61,373.35

**Comment by MikeFultonDev** [comment]
@IgorTodorovskiIBM can you confirm this is fixed now? Looks like it was an issue in auto-conversion (bash?) and not iconv?

**Comment by IgorTodorovskiIBM** [comment]
Yes, this was caused by double-conversion, address in https://github.com/zopencommunity/libiconvport/pull/35. @gngrossi ok to close?

**Comment by gngrossi** [comment]
Looks good...closing
thanks

[SYSA] bash-5.2$ etoa TBASVC_04022025.csv
/hewitt/zopentools/zopen_repo/usr/local/altbin/iconv

[SYSA] bash-5.2$ ls -lT *
t ISO8859-1   T=on  -rw-r-----   1 @02858   @ISCICS1 3186542 Apr  4 07:28 TBASVC_04022025.csv
t IBM-1047    T=on  -rw-r-----   1 @02858   @ISSTC00 3186542 Apr  4 07:28 TBASVC_04022025.csv.ebcdic
t IBM-1047    T=on  -rwxr-x---   1 @02858   @ISCICS1     328 Apr  4 07:50 etoa

[SYSA] bash-5.2$ diff TBASVC_04022025.csv TBASVC_04022025.csv.ebcdic

[SYSA] bash-5.2$ head TBASVC_04022025.csv TBASVC_04022025.csv.ebcdic
==> TBASVC_04022025.csv <==
"DATE","TRANNAME","CLIENTID","TBAIND","TBAS01","cputm","ttrans","cpuptrn","db2reqct","mqreqct",DB2/tran,SUs/tran,TotalCost
"01APR25","Q0AG",00095,101,"PRTPRMDATA",7259.3,499559,0.0145,246836686,1144944,494.11,129.60,217.78

==> TBASVC_04022025.csv.ebcdic <==
"DATE","TRANNAME","CLIENTID","TBAIND","TBAS01","cputm","ttrans","cpuptrn","db2reqct","mqreqct",DB2/tran,SUs/tran,TotalCost
"01APR25","Q0AG",00095,101,"PRTPRMDATA",7259.3,499559,0.0145,246836686,1144944,494.11,129.60,217.78
